nibabel/funcs.py

```diff
@@ -115,15 +117,19 @@ def concat_images(images, check_affines=True):
     affine = img0.affine
     header = img0.header
     out_shape = (n_imgs, ) + i0shape
```

Comment:
Got it.  I have an idea how to do all the cases efficiently; I'll try it out sometime today.  I'll also look into improving the test coverage and checks.